update connect/ca's vault AuthMethod conf section (#16346)
hashicorp/consul: website/content/docs/connect/ca/vault.mdx

  on how to configure individual auth methods. If auth method is provided, Consul will obtain
  a new token from Vault when the token can no longer be renewed.

   - `Type`/ `type` (`string: ""`) - The type of Vault auth method. Valid options are "approle", "aws", "azure", "gcp", "jwt" and "kubernetes".

   - `MountPath`/ `mount_path` (`string: <AuthMethod.Type>`) - The mount path of the auth method.
      If not provided the auth method type will be used as the mount path.

>    - `Params`/`params` (`map: nil`) - The parameters to configure the auth method. The configuration parameters needed will depend on which auth type you are using. Please see the Vault Agent, auto-auth method documentation for details on their configuration options: [AppRole](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/approle#configuration), [AWS](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/aws#configuration), [Azure](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/azure#configuration), [GCP](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/gcp#configuration), [JWT](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/jwt#configuration), [Kubernetes](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/kubernetes#configuration).

Reviewer: ```suggestion
   - `Params`/`params` (`map: nil`) - The parameters to configure the auth method. The configuration parameters needed will depend on which auth type you are using. Please refer to the Vault Agent auto-auth method documentation for details on their configuration options: [AppRole](/vault/docs/agent/autoauth/methods/approle#configuration), [AWS](/vault/docs/agent/autoauth/methods/aws#configuration), [Azure](/vault/docs/agent/autoauth/methods/azure#configuration), [GCP](/vault/docs/agent/autoauth/methods/gcp#configuration), [JWT](/vault/docs/agent/autoauth/methods/jwt#configuration), [Kubernetes](/vault/docs/agent/autoauth/methods/kubernetes#configuration).
```

>    ~> **Note:** Only the authentication related fields are supported, not the optional management fields. E.g. JWT's `path` and `role` are supported but not its `remove_jwt_after_reading` field.

Reviewer: ```suggestion
   Only the authentication related fields (for example, JWT's `path` and `role`) are supported. The optional management fields (for example: `remove_jwt_after_reading`) are not supported.
```
